DEVELOPERS.md

```diff
@@ -0,0 +1,106 @@
+# Trillian: Information for developers
+
+ - [Overview](#overview)
+ - [Working on the Code](#working-on-the-code)
+     - [Rebuilding Generated Code](#rebuilding-generated-code)
+     - [Updating Dependencies](#updating-dependencies)
+     - [Running Codebase Checks](#running-codebase-checks)
+
+## Overview
+
+If you'd like to help us work on Trillian then this is the document for you!
+
+We'd love to have your help to make Trillian better for everyone, and welcome
+PRs and Issues for everything from typos to adding new documentation, small bug
+fixes through to new features (although if you're doing the latter, then please
+do [get in touch with us](https://github.com/google/trillian#support)
+beforehand to ensure that work isn't being duplicated).
+
+## Working on the Code
+
+Developers who want to make changes to the Trillian codebase need some
+additional dependencies and tools, described in the following sections.  The
+[Travis configuration](.travis.yml) for the codebase is also useful reference
+for the required tools and scripts, as it may be more up-to-date than this
+document.
+
+### Rebuilding Generated Code
+
+Some of the Trillian Go code is autogenerated from other files:
+
+ - [gRPC](http://www.grpc.io/) message structures are originally provided as
+   [protocol buffer](https://developers.google.com/protocol-buffers/) message
+   definitions.
+ - Some unit tests use mock implementations of interfaces; these are created
+   from the real implementations by [GoMock](https://github.com/golang/mock).
+ - Some enums have string-conversion methods (satisfying the `fmt.Stringer`
+   interface) created using the
+   [stringer](https://godoc.org/golang.org/x/tools/cmd/stringer) tool (`go get
+   golang.org/x/tools/cmd/stringer`).
+
+Re-generating mock or protobuffer files is only needed if you're changing
+the original files; if you do, you'll need to install the prerequisites:
+
+  - `mockgen` tool from https://github.com/golang/mock
+  - `stringer` tool from https://golang.org/x/tools/cmd/stringer
+  - `protoc`, [Go support for protoc](https://github.com/golang/protobuf),
+     [grpc-gateway](https://github.com/grpc-ecosystem/grpc-gateway) and
+     [protoc-gen-doc](https://github.com/pseudomuto/protoc-gen-doc).
+  - protocol buffer definitions for standard Google APIs:
+
+    ```bash
+    git clone https://github.com/googleapis/googleapis.git $GOPATH/src/github.com/googleapis/googleapis
```

Comment:
Does this still work in the "$GOPATH is deprecated" world we now find ourselves in, where Trillian might not be under $GOPATH?